pynpoint/processing/psfpreparation.py

```diff
@@ -123,32 +131,40 @@ def run(self) -> None:
 
         # Run the PSFpreparationModule for each subset of frames
         start_time = time.time()
-        for i, _ in enumerate(frames[:-1]):
+        for j in range(wavelength_dim):
+            for i, _ in enumerate(frames[:-1]):
 
-            # Print progress to command line
-            progress(i, len(frames[:-1]), 'Preparing images for PSF subtraction...', start_time)
+                # Print progress to command line
+                progress(i + j * len(frames[:-1]), wavelength_dim * len(frames[:-1]),
+                         'Preparing images for PSF subtraction...', start_time)
 
-            # Get the images and ensure they have the correct 3D shape with the following
-            # three dimensions: (batch_size, height, width)
-            images = self.m_image_in_port[frames[i]:frames[i+1], ]
+                # Get the images and ensure they have the correct 3D shape with the following
+                # three dimensions: (batch_size, height, width)
+                if len(im_shape) < 4:
+                    images = self.m_image_in_port[frames[i]:frames[i+1], ]
+                else:
+                    images = self.m_image_in_port[j, frames[i]:frames[i+1], ]
 
-            if images.ndim == 2:
-                warnings.warn('The input data has 2 dimensions whereas 3 dimensions are required. '
-                              'An extra dimension has been added.')
+                if images.ndim == 2:
+                    warnings.warn('The input data has 2 dimensions whereas 3 dimensions are required. '
+                                  'An extra dimension has been added.')
 
-                images = images[np.newaxis, ...]
+                    images = images[np.newaxis, ...]
 
-            # Apply the mask, i.e., set all pixels to 0 where the mask is False
-            images[:, ~mask] = 0.
+                # Apply the mask, i.e., set all pixels to 0 where the mask is False
+                images[:, ~mask] = 0.
```

Comment:
In this case, the same mask is applied to all wavelengths? Should the size not increase linearly with wavelengths such that the masks overlap after the scaling in the PCA module?

Reply:
In the case of a non-scaling mask, the inner and outer radii are limited by the scaling of the PSF. If you cut to close around a possible planet, the PSF model will no longer be accurate. In the case of a scaling mask, the inner and outer radii are limited by the "radial displacements" of a possible exoplanet. If you cut to close around a possible planet, you'll mask out the planet in some wavelengths.

Since the scaling is the same in both cases, the area which should not be masked out is the same in both cases. Additionally, masking a stationary image instead of a scaling one is more intuitive and therefore easier to handle. I would therefore suggest leaving it as it is.

Reply:
OK